indico/modules/events/models/events.py

```diff
@@ -1110,6 +1110,16 @@ def is_user_speaker(self, user):
                 ))
                 .has_rows())
 
+    def _get_email_templates_rules(self):
+        rules = {}
+        for template in self.abstract_email_templates:
+            for rule in template.rules:
+                for key, value in rule.items():
+                    if not rules.get(key):
+                        rules[key] = set()
+                    rules[key].update(value)
+        return rules.get('state', [])
+
```

Comment:
This can be simplified a bit. Also, I would not add this to the event model, unless it's needed in Jinja. Especially with a generic name like this that doesn't make it clear that this is *only* for abstracts. This could go into something like `events/abstracts/notifications` or utils.

```suggestion
        rules = set()
        for template in self.abstract_email_templates:
            for rule in template.rules:
                if state := rule.get('state'):
                    rules.update(state)
        return rules
```